lib/smartcar/utils.rb

```diff
@@ -13,43 +13,104 @@ def initialize(options = {})
       end
     end
 
-    # Utility method to return a hash of the isntance variables
-    #
-    # @return [Hash] hash of all instance variables
-    def to_hash
-      instance_variables.each_with_object({}) do |attribute, hash|
-        hash[attribute.to_s.delete('@').to_sym] = instance_variable_get(attribute)
-      end
-    end
-
     # gets a given env variable, checks for existence and throws exception if not present
     # @param config_name [String] key of the env variable
     #
     # @return [String] value of the env variable
     def get_config(config_name)
-      config_name = "INTEGRATION_#{config_name}" if ENV['MODE'] == 'test'
+      # ENV.MODE is set to test by e2e tests.
+      config_name = "E2E_#{config_name}" if ENV['MODE'] == 'test'
       raise Smartcar::ConfigNotFound, "Environment variable #{config_name} not found !" unless ENV[config_name]
 
       ENV[config_name]
     end
 
-    # Given the response from smartcar API, returns an error object if needed
+    def build_meta(headers)
+      meta_hash = {
+        'sc-data-age' => :data_age,
+        'sc-unit-system' => :unit_system,
+        'sc-request-id' => :request_id
+      }.each_with_object({}) do |(header_name, key), meta|
+        meta[key] = headers[header_name] if headers[header_name]
+      end
+      meta = JSON.parse(meta_hash.to_json, object_class: OpenStruct)
+      meta.data_age &&= DateTime.parse(meta.data_age)
+
+      meta
+    end
+
+    def build_response(body, headers)
+      response = JSON.parse(body.to_json, object_class: OpenStruct)
+      response.meta = build_meta(headers)
+      response
+    end
+
+    def build_aliases(response, aliases)
+      (aliases || []).each do |original_name, alias_name|
+        response.send("#{alias_name}=".to_sym, response.send(original_name.to_sym))
+      end
```

Comment:
does this need to delete the original name property as well? The way I'm understanding this the resulting object would have the same value available under both keys

Reply:
Yes, it will have under both keys. It was purposeful , the approach was that since the language uses snake case as "best practice" we will define the snake_case accessors but for some reason . But if a developer choses to ignore that standard (which i have seen among teams with multi language microservices) then they will still have an option of using the camelCase ones which is close to the interface documented.

Reply:
ah okay, got it